ReactWindows/ReactNative.Shared/UIManager/UIViewOperationQueue.cs

```diff
@@ -480,19 +481,16 @@ private void OnRendering(object sender, object e)
                 }
             }
 
+            Action[] actions = EMPTY_ACTIONS_ARRAY;
             lock (_gate)
             {
-                try
-                {
-                    foreach (var batch in _batches)
-                    {
-                        batch();
-                    }
-                }
-                finally
-                {
-                    _batches.Clear();
-                }
+                actions = _batches.ToArray();
```

Comment:
We could avoid GC pressure by using a secondary list field (`_clonedBatches`) that is accessed only in the OnRendering callback (so no lock needed), where we copy from `_batches` to `_clonedBatches` (`_clonedBatches.AddRange(_batches)`).  I believe `.Clear` on `List<T>` does not "shrink" the inner array, so we would effectively keep two lists with size matching the largest UI batch that has occurred in the application.  Make sense?

---
In reply to: [88969646](https://github.com/ReactWindows/react-native-windows/pull/891#discussion_r88969646) [](ancestors = 88969646,88969471,88968974)